adapters/redis/lib/opentelemetry/adapters/redis/patches/client.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Redis
+      module Patches
+        # Module to prepend to Redis::Client for instrumentation
+        module Client
+          def call(*args, &block)
+            response = nil
+
+            tracer.in_span(
+              Utils.format_command(args),
+              attributes: client_attributes.merge(
+                'db.statement' => Utils.format_statement(args)
+              ),
+              kind: :client
+            ) do |_span|
+              response = super(*args, &block)
+            end
+
+            response
+          end
+
+          def call_pipeline(*args, &block)
+            response = nil
+
+            tracer.in_span(
+              'pipeline',
+              attributes: client_attributes.merge(
+                'db.statement' => Utils.format_pipeline_statement(args)
+              ),
+              kind: :client
+            ) do |_span|
+              response = super(*args, &block)
+            end
+
+            response
+          end
+
+          private
+
+          def client_attributes
+            host = options[:host]
+            port = options[:port]
+
+            {
+              'component' => 'redis',
+              'db.type' => 'redis',
+              'db.instance' => options[:db].to_s,
+              'db.url' => "redis://#{host}:#{port}",
+              'net.peer.name' => host,
+              'net.peer.port' => port
+            }
+          end
```

Comment:
It looks like `db` is changeable at runtime, but is everything else going to be static data once the client is initialized? If so, can we memoize and freeze this data instead of making it each time? The `db.name` can be merged in with `db.statement` in call.

Reply:
> is everything else going to be static data once the client is initialized?

@mwear I don't think so. https://github.com/redis/redis-rb#sentinel-support and https://github.com/redis/redis-rb#cluster-support seem to indicate to me that the port and even host can be different between requests. Using the example code I was able to see different hosts/ports for the spans using Sentinel. Unfortunately it does not appear that `fakeredis` which we are using for the tests supports Sentinel or clusters.

> The db.name can be merged in with db.statement in call.

Can you explain this more?

Reply:
I was just saying that if the config was static aside from `db.name` we could have a frozen hash and merge('db.name' => ..., 'db.statement') with it. It's a moot point given that config will change at runtime. Thanks for your research!